### ISSUE TITLE
Fonts not changing on firefox

### ISSUE THREAD

**Issue by rulet** [darkreader/darkreader#5673]
# Bug Report

## Describe the bug
Fonts not changing on firefox

## Expected behavior
changing font

## System Information

- OS: Windows 10
- Browser: Firefox 78.10.0esr
- Dark Reader Version:

**Comment by Gusted** [comment]
Hi @rulet!

Could you elaborate a bit what your bug is about?

Regards,
Gusted

**Comment by rulet** [comment]
Dark Reader extension doesn't change font  on any website.

**Comment by Gusted** [comment]
Hmm, could you try a common font like "sans-serif". It could be that the font isn't installed.

**Comment by rulet** [comment]
The font is installed, extension works under chrome, I mean extension changes font under chrome.

**Comment by rulet** [comment]
I made clean install of firefox esr 78.10.0 and disabled all other extensions and I may confirm that Dark Reader extension doesn't change font, I tried also to change to "sans-serif" and fonts doesn't change on any webpage.

**Comment by Gusted** [comment]
Hmm that's very weird it almost seems like their is something going on with ESR version, I will have a check with the ESR version.

**Comment by rulet** [comment]
Any news?

**Comment by Gusted** [comment]
Other than I wasn't able to reproduce and likelwise we've never had such bug report and couldn't find anything either on it. Not really some new news.

**Comment by AndrewDaGuy** [comment]
> I made clean install of firefox esr 78.10.0 and disabled all other extensions and I may confirm that Dark Reader extension doesn't change font, I tried also to change to "sans-serif" and fonts doesn't change on any webpage.

Can't reproduce this either, can you try live booting something like [Linux Mint](https://linuxmint.com/) on a USB then try to reproduce it? (It should have Firefox preinstalled)

**Comment by rulet** [comment]
Usual firefox updates works, but not working on esr. Actually I don't need fonts management by dark reader as firefox can do it itself, and also there are some bugs with dark reader showing some web elements when using dark reader font management.

**Comment by AndrewDaGuy** [comment]
> dark reader not showing some web elements

Yea, it seems like its also changing the font awesome / icon fonts to something like sans-serif and turns it into something like this:
![image](https://user-images.githubusercontent.com/66189242/117133947-7420be00-ad94-11eb-8c76-ea74cdb74fc7.png)

**Comment by rulet** [comment]
> something like this:

Yes, that's what I'm talking about, and it concerns others web elements too.

**Comment by Gusted** [comment]
Those bugs are beyond our scope as the font your using haven't implemented those characters. We simply provide the CSS to set such font and the rest is beyond our scope. Il suggest having installed a extension like stylus and try to write a simple CSS
```CSS
* {
    font-family: sans-serif !important;
}
```
and see if that works as like I said we provide the font and if the browser won't display it or not render it correctly that's not our problem nor our ability to fix.

**Comment by mickey86** [comment]
Hello,

I was also concerned by this weird thing on firefox ESR: no font substitution.

I searched a little in firefox’s configuration what would block font substitution. I found on about:config page: browser.display.use_document_fonts was valued with 1, and after I turned it to 0, the font substitution worked immediately.

My 2 cents…

**Comment by Gusted** [comment]
> browser.display.use_document_fonts was valued with 1, and after I turned it to 0, the font substitution worked immediately.

That's one of the best 2 cents I've ever seen. And after a quick search in firefox's code I could find the reason of this preference -> see the comment https://searchfox.org/mozilla-central/source/servo/components/style/properties/cascade.rs#853-887 
Whereby this preference was introduced to disable user fonts -> https://bugzilla.mozilla.org/show_bug.cgi?id=789788

I think it's also most likely the case with this issue.

@rulet if you would be so nice, to set `browser.display.use_document_fonts` to 0 on the `about:config` page.

**Comment by rulet** [comment]
> I found on about:config page: browser.display.use_document_fonts was valued with 1, and after I turned it to 0, the font substitution worked immediately.

I tried that, but that didn't work.

**Comment by mickey86** [comment]
> > I found on about:config page: browser.display.use_document_fonts was valued with 1, and after I turned it to 0, the font substitution worked immediately.
> 
> I tried that, but that didn't work, firefox 78.10.1esr. browser.display.use_document_fonts setting only changes this toggle here(so, no need to use about:config for this):
> ![Scr](https://user-images.githubusercontent.com/2847261/119742011-8e2f5700-be8f-11eb-87b6-b88c0625a51b.png)
> but has no influence on dark reader extension behavior.

You’re right, I also noted the option is available in preferences :)

Hum… Before having the font substitution working, I played with the dark reader extension’s settings… No other idea for the moment.

**Comment by haolly** [comment]
still not works

**Comment by d3vilguard** [comment]
+1 for it not working. I'm desperately trying to apply a font through dark reader in firefox. I have my font placed in:
/usr/share/fonts, ./local/share/fonts, $HOME/.fonts and it just fails and fails. I have done it before on firefox. No idea why it suddenly started to fail doing it. It's like it doesn't even find the font. Doesn't list it, typing it in manually doesn't work either. Both chrome and firefox find the font I'm trying to apply.

**Comment by d3vilguard** [comment]
This could be a general firefox bug. Making a new profile and using it has all the fonts:
https://support.mozilla.org/en-US/questions/1355994
